app/main/lib/shared_models/shared_model.py

```diff
@@ -12,141 +12,159 @@
-    class_ = SharedModel.import_model_class(model_class)
-    return class_(model_key, options)
-
-  def __init__(self, model_key, options={}):
-    self.options = options
-    self.queue_name = model_key
-    self.datastore = redis.Redis(host=app.config['REDIS_HOST'], port=app.config['REDIS_PORT'], db=app.config['REDIS_DATABASE'])
-
-  def get_task(self, timeout=0):
-    item = self.datastore.blpop(self.queue_name, timeout)
-    if item:
-      task = Task(**json.loads(item[1].decode('utf-8')))
-      app.logger.info('[%s] Picking up task %s', self.queue_name, task.task_id)
-      return task
-
-  def send_response(self, task, response):
-    self.datastore.set(task.task_id, json.dumps({"response": response}))
-
-  def run(self):
-    while True:
-      task = self.get_task()
-      if task:
-        self.send_response(
-          task,
-          self.respond(task.task_package)
+    @staticmethod
+    def import_model_class(model_class):
+        class_name = re.sub(r"(?<!^)(?=[A-Z])", "_", model_class).lower()
+        return getattr(
+            importlib.import_module("app.main.lib.shared_models.%s" % class_name),
+            model_class,
+        )
+
+    @staticmethod
+    def start_server(model_class, model_key, options={}):
+        class_ = SharedModel.import_model_class(model_class)
+        instance = class_(model_key, options)
+        instance.load()
+        app.logger.info("[%s] Serving model...", model_key)
+        r = redis.Redis(
+            host=app.config["REDIS_HOST"],
+            port=app.config["REDIS_PORT"],
+            db=app.config["REDIS_DATABASE"],
+        )
+        r.set(
+            "SharedModel:%s" % model_key,
+            json.dumps(
+                {"model_class": model_class, "model_key": model_key, "options": options}
+            ),
+        )
+        r.sadd("SharedModel", model_key)
+        instance.bulk_run()
+
+    @staticmethod
+    def get_servers():
+        r = redis.Redis(
+            host=app.config["REDIS_HOST"],
+            port=app.config["REDIS_PORT"],
+            db=app.config["REDIS_DATABASE"],
+        )
+        return [server.decode("utf-8") for server in r.smembers("SharedModel")]
+
+    @staticmethod
+    def get_client(model_key, options={}):
+        r = redis.Redis(
+            host=app.config["REDIS_HOST"],
+            port=app.config["REDIS_PORT"],
+            db=app.config["REDIS_DATABASE"],
+        )
+        model_info = r.get("SharedModel:%s" % model_key)
+        assert model_info is not None, f"Unable locate model info for key {model_key}"
+        model_class = json.loads(model_info.decode("utf-8"))["model_class"]
+        class_ = SharedModel.import_model_class(model_class)
+        return class_(model_key, options)
+
+    def __init__(self, model_key, options={}):
+        self.options = options
+        self.queue_name = model_key
+        self.datastore = redis.Redis(
```

Comment:
Again, there's a ton of redis.Redis in this file - feels like this could be DRY'ed up a bit